### ISSUE TITLE
Players spawn in the wrong location after death

### ISSUE THREAD

**Issue by HelenaKitty** [cuberite/cuberite#3611]
Client version:   1.9
Server OS:        Linux
Commit id:        83fa6cb851536839ebd85b1e91fda928f47eb674

### Expected behavior

Players should spawn according to /spawn, or some kind of death spawnpoint (if implemented)

### Actual behavior

Instead, players spawn in the wrong place, also in a place I'd like to keep private.

### Steps to reproduce the behavior

Kill yourself

**Comment by madmaxoft** [comment]
Players respawn at their last bed position, or if that is not available, at the world's spawn. What else would you expect?

**Comment by HelenaKitty** [comment]
I was spawned into the middle of an underground railway network which is not the spawnpoint, nor is there a bed.

**Comment by spekdrum** [comment]
Im respawning in the roof of my house instead of my bed when I die.

**Comment by mathiascode** [comment]
@bearbin Did you reproduce the issue?

**Comment by bearbin** [comment]
On Mon, 18 May 2020 06:07:56 -0700
Mat <notifications@github.com> wrote:

> @bearbin Did you reproduce the issue?
> 

Nope, I just added the label because ingame testing is needed to
identify exactly what the issue (is/was) here.

**Comment by tigerw** [comment]
Seeing this issue pop up in notifications be like

![Déjà vu](https://i.imgflip.com/ygft8.jpg)

**Comment by JK2Kgit** [comment]
I did a few checks and after setting a respawn point by sleeping in bed then destroying that bed. Spawnpoint didn't got reset to world spawn. I still spawned where the bed previously was. My guess is that there is something wrong in check to weather or not bed is available

**Comment by madmaxoft** [comment]
I believe that's the vanilla behavior as well (or at least it used to be) - sleeping on a bed sets the spawnpoint, but destroying that bed doesn't un-set the spawnpoint.

**Comment by Janmm14** [comment]
> 
> 
> I believe that's the vanilla behavior as well (or at least it used to be) - sleeping on a bed sets the spawnpoint, but destroying that bed doesn't un-set the spawnpoint.

The vanilla behaviour is documented here:
https://minecraft.fandom.com/wiki/Bed#Setting_the_spawn_point

tldr: On respawn it checks whether there is bed and a place to spawn the player in a couple blocks around the bed, otherwise the world spawn is used.
